robber/matchers/numbers.py

```diff
@@ -34,10 +34,35 @@ def matches(self):
     def failure_message(self):
         return 'Expected %g to be within %g and %g' % (self.actual, self.expected, self.args[0])
 
+
+class Change(Base):
+    def __init__(self, callable, obj=None, *args):
+        self.callable = callable
+        self.obj = obj
+        self.args = args
+        self.message = None
+
+    def match(self):
+        return self
+
+    def by(self, amount=0):
+        changed = self.callable(self.obj) - self.obj
+
+        if changed != amount:
+            message = self.message or self.failure_message(self.callable.__name__, self.obj, amount, changed)
+            raise BadExpectation(message)
+
+        return expect(self.obj)
+
+    def failure_message(self, callable_name, obj, changed, got):
+        return 'Expect function %s to change %g by %g, but was changed by %g' % (callable_name, obj, changed, got)
```

Comment:
Hm, not very sure but I think it's good enough. Just keep it and consider it later then.